### PR TITLE
Testing Your Ruby Code - Enumerable

### PR DESCRIPTION
**Pull request by IjayAbby**
In this project requirement, I have completed the following:
- Add specs for all Enumerable methods I previously created.
- Fix rubocop errors
- Ensure that all tests are passing.

### CODE REVIEW

**Comment by mimipeshy** [comment]
### STATUS : CHANGES REQUIRED :recycle: 

Hi @IjayAbby :wave:  :wave: ,good job on the test cases, they pass and you should be proud of yourself :clap:  :clap: . However, before approving the project lets make a few changes listed below :arrow_down: 

- [ ] Add more tests for each method an example for #my_any
- You can check if it returns an enumerable if the block is not given.
- When a range is passed as an argument, it returns a range of elements
**Note:** Test for everything the method does and see if it passes or fails.

- [ ] Add tests for `multiply_els`

Thats all :relieved:  kindly submit another review when done.

Happy Coding  :computer:  :technologist: 

Regards
Peris Ndanu
Feel free to slack me if you have any questions (perisndanu)

File: enumerables.rb [hunk 1]
```diff
@@ -152,7 +152,6 @@ def multiply_els(array)
 range = Range.new(5, 50)
 
 p range.my_each_with_index(&block)
-
```
Comment (descholar-ceo) [inline enumerables.rb]: Remove this code, they are for debugging purpose

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Change Required :vertical_traffic_light: :warning: 
## Hello @IjayAbby  :handshake: :handshake: :handshake: 
:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
## Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes]
- [x] You have implemented only positive cases, as the previous reviewer said, try to test even negative cases. For example for the `#my_each` method, you should have tested when a block is not given and check if it returns an enumerator like the following code snippet
```ruby
  it 'When no block given, my_each method return an enumerator' do
      expect(my_array.my_each).to be_an(Enumerator)
    end
```
:point_right: Try to implement negative cases on all methods

- [x] You should refactor the readme file to add how to run tests

- [x] Remove all debuging codes like the `p` statements which are left from your codes

- [x] Edit this Pull Request's name, make it descriptive and meaningful. I mean that it should reflect the work you are implementing on this Pull Request, currently its name is `code review` => that name doesna't reflect the actual work that you are implementing in it.

If anything seems to be unclear to you, don't hesitate to contact me for more clarification slack: `@Mugirase Emmanuel` | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Review by arslanbisharat — APPROVED** [review]
# Approved :smile: 

Hello Developer :wave:

I just reviewed your project and I am very happy to inform you that your project is approved.
Congratulations on achieving this milestone in your career. :clap:

_I am looking forward to seeing your next project._

Stay safe and give me a smile 😇

### Happy Coding
------------------------------------------------------------------------------------------------
**Regards,
Arslan Bisharat
slack: arslanbisharat**

**Pull request merged**